internal/dbunique/unique_fields.go

```diff
@@ -0,0 +1,123 @@
+package dbunique
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/riverqueue/river/rivertype"
+	"github.com/tidwall/gjson"
+)
+
+var (
+	// uniqueFieldsCache caches the unique fields for each JobArgs type. These are
+	// global to ensure that each struct type's tags are only extracted once.
+	uniqueFieldsCache = make(map[reflect.Type][]string) //nolint:gochecknoglobals
+	cacheMutex        sync.RWMutex                      //nolint:gochecknoglobals
+)
+
+type jobArgs interface {
+	Kind() string
+}
```

Comment:
Oops, don't need both this _and_ the `rivertype.JobArgs` duplicate definition. Lmk if you think this unexported copy is the right way to go.